src/testing/cluster.zig

```diff
@@ -460,6 +460,13 @@ pub fn ClusterType(comptime StateMachineType: anytype) type {
                 var advanced = false;
                 advanced = cluster.network.step() or advanced;
 
+                for (cluster.clients, cluster.client_eviction_reasons) |*client, eviction_reason| {
+                    if (client.* != null and eviction_reason != null) {
+                        client.*.?.deinit(cluster.allocator);
+                        client.* = null;
+                    }
+                }
```

Comment:
We no longer can `client.deinit` right from the eviction callback, so that now sets a flag (via eviciton_reason), and the actual deinit happens here.